contracts/ExampleOfferResolver.cdc

```
@@ -0,0 +1,64 @@
+import Resolver from "./Resolver.cdc"
+import NonFungibleToken from "./utility/NonFungibleToken.cdc"
+import MetadataViews from "./utility/MetadataViews.cdc"
+
+pub contract ExampleOfferResolver {
+
+    pub let ExampleOfferResolverStoragePath: StoragePath
+
+    /// Resolver resource holds the Offer exchange resolution rules.
+    pub resource OfferResolver: Resolver.ResolverPublic {
+        /// checkOfferResolver
+        /// Holds the validation rules for resolver each type of supported ResolverType
+        /// Function returns TRUE if the provided nft item passes the criteria for exchange
+        pub fun checkOfferResolver(
+         item: &AnyResource{NonFungibleToken.INFT, MetadataViews.Resolver},
+         offerFilters: {String: AnyStruct}
+         ): Bool {
+
+            let resolver = offerFilters["resolver"]! as? UInt8 ?? panic("resolver value is missing or non UInt8 resolver")
+            switch resolver {
+
+                case Resolver.ResolverType.NFT.rawValue:
+                    let nftId = offerFilters["nftId"]! as? UInt64 ?? panic("nftId value is missing or non UInt64 nftId")
+                    assert(item.id == nftId, message: "item NFT does not have specified ID")
+                    return true
+
+                case Resolver.ResolverType.MetadataViews.rawValue:
+                    let views = item.resolveView(Type<MetadataViews.Editions>()) 
```

IMHO this should not panic. It should return and decline this filter

---

True, I agree with this but on giving a second thought, it would be better to tell the end user the exact cause of failure instead of just failing silently, If you think otherwise then I am happy to change this.